src/library/dlgtrackinfo.cpp

```diff
@@ -483,6 +487,20 @@ void DlgTrackInfo::saveTrack() {
         return;
     }
 
+    // In case Apply is triggered by hotkey AND a QLineEdit with pending changes
+    // is focused AND the user did not hit Enter to finish editing,
+    // the content of that focused line edit would be reset to the last confirmed state.
+    // This hack emits the editingFinished() signal for the focused QLineEdit.
+    if (this == QApplication::activeWindow()) {
+        auto focusWidget = QApplication::focusWidget();
+        if (focusWidget) {
+            QLineEdit* focusedLineEdit = static_cast<QLineEdit*>(focusWidget);
```

Comment:
This is undefined behavior if the focused widget is not of type `QLineEdit`.

Reply:
See [`qobject_cast`](https://doc.qt.io/qt-5/qobject.html#qobject_cast-1) for the safe cast.

Reply:
Is it possible to switch focus to the Apply button and then process all subsequent events before continuing? This would be a more robust solution because that is independent of the actual widgets that are used.

Reply:
> See [`qobject_cast`](https://doc.qt.io/qt-5/qobject.html#qobject_cast-1) for the safe cast.

Yeah, thanks!

Reply:
> Is it possible to switch focus to the Apply button and then process all subsequent events before continuing?

_The_ benefit of using hotkeys is that the metadata field remains focused when switching tracks, see the mass-edit example I mentioned.

alternative hacks:
* ~~store focus widget, focus Apply button to fire editingFinished, re-focus previous widget, but this is a bit over the top to fix the behaviour~~
* ~~send a `Return` key event
requires filtering for non-QPushButton widgets~~
* ~~sending `Tab` then `Shift+Tab`
that's is even more of a hack~~
* ~~have a list of all used QLineEdits (they're hardcoded here anyway) and in `saveTrack()` iterate over that to emit editingFinished() for _all_ of them~~
* shortest widget-independent and most straight-forward hack:
```
auto focusWidget = QApplication::focusWidget();
if (focusWidget) {
    focusWidget->clearFocus();
    focusWidget->setFocus();
}
```
 : )

Reply:
I switched to `qobject_cast`, please let me know if that is sufficient for now.

Reply:
Didn't the widget-independent solution work? This would be much more robust and cleaner.

Reply:
It works, but still it's a hack -- and I thought we prefer 'official ways' over hacks.
I don't care as long as the bug is fixed.

Reply:
Leaving the focused widget is what is supposed to happen before hitting Apply. That's the regular flow. I don't like to introduce shortcuts that work differently.

Reply:
The focus change variant works for me. Didn't notice that you introduced it here.

Reply:
> Leaving the focused widget is what is supposed to happen before hitting Apply

...when you use the mouse, sure.
> I don't like to introduce shortcuts that work differently


They **do** work differently: QShortcut makes its QPushButton emit `clicked()` but it does not force-focus the button.
FWIW #3906 broke the flow for hotkey users, or at least made it more complex by requiring to hit Return.
(no, that's no occassion to post that famous xkcb cartoon... ;)

Reply:
> The focus change variant works for me. Didn't notice that you introduced it here.

I just did after you asked for it.